src/main/java/util/JsonFileConverter.java

```diff
@@ -0,0 +1,85 @@
+package util;
+
+import com.google.gson.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.*;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Optional;
+
+/**
+ * Converts JSON file from and to object
+ */
+public class JsonFileConverter {
+    private static final Logger logger = LogManager.getLogger(JsonFileConverter.class.getName());
+    private static final String CHARSET = "UTF-8";
+
+    private Gson gson;
+    private final File jsonFile;
+
+    public JsonFileConverter(File jsonFile) {
+        this.jsonFile = jsonFile;
+        setupGson();
+    }
+
+    /**
+     * Save a given object into JSON file
+     * @param object object to be saved to JSON file
+     */
+    public void saveToFile(Object object) throws IOException {
+        createDirectories();
+        Writer writer = new OutputStreamWriter(new FileOutputStream(jsonFile) , CHARSET);
+        gson.toJson(object, object.getClass(), writer);
+        writer.close();
+    }
+
+    /**
+     * Loads an object from a JSON file
+     * @param objectClass class of the object to be loaded; used to parse the JSON
+     * @return The object from JSON file, or Optional.empty() if the file can't be parsed
+     */
+    public Optional<Object> loadFromFile(Class<?> objectClass) {
```

Comment:
Could we use `public <T> Optional<T> loadFromFile(Class<T> objectClass) {`? Then we don't have to cast down at use sites.